lightning/src/ln/channelmanager.rs

```diff
@@ -3118,13 +3206,37 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 													};
 													match next_hop {
 														onion_utils::Hop::Receive(hop_data) => {
-															match self.construct_recv_pending_htlc_info(hop_data, incoming_shared_secret, payment_hash, amt_to_forward, outgoing_cltv_value, Some(phantom_shared_secret)) {
+															match self.construct_recv_pending_htlc_info(hop_data, forward_info.incoming_shared_secret, forward_info.payment_hash, forward_info.amt_to_forward, forward_info.outgoing_cltv_value, Some(phantom_shared_secret)) {
 																Ok(info) => phantom_receives.push((prev_short_channel_id, prev_funding_outpoint, vec![(info, prev_htlc_id)])),
 																Err(ReceiveError { err_code, err_data, msg }) => fail_forward!(msg, err_code, err_data, Some(phantom_shared_secret))
 															}
 														},
 														_ => panic!(),
 													}
+												} else if forward_info.amt_incoming.is_some() && fake_scid::is_valid_intercept(&self.fake_scid_rand_bytes, short_chan_id) {
```

Comment:
I thought of a issue potential attack vector here that I wanted to raise in the review club earlier today, but we didnät have time to go over the question about such vectors:

Shouldn't we check here that the `fake_scid` is a `scid` that we actually have generated, and not only that it's within the Intercept namespace?

Else, if another peer is able to generate a `fake_scid` that we determine to be in the Intercept namespace but that we didn't generate ourselves, we face 2 issues:

1. Since the `PaymentIntercepted` event is serialized for peers we have no active channel to (and isn't dropped on disconnect like other events for that type of peers), as well as `PendingInterceptedHTLC` which aren't handled, random peers we have no channels to could potentially be able cause extensive serialization of huge amount of events/structs.

2. Probably even worse, just one successful `fake_scid` attack by a random peer that passes the `is_valid_intercept`, will serialize the `PaymentIntercepted` event + `PendingInterceptedHTLC` for any LDK user, even users that didn't explicitly choose to use the new Intercept HTLC feature. That will break the backwards compatibility for such users, hence opening up for a backwards compatibility attack.

Looking at the implemenation for `fake_scids` + `namespaces`, it shouldn't be too expensive for random peers to brute force a single `fake_scid` that we will accept as a valid `Intercept` `fake_scid`? 

I think we therefore need to also add add a way to track which `fake_scids` we have actually generated to not face this issue? I guess one alternative solution to this would also be to add a config flag for enabling `Intercept HTLC `support, which defaults to `false`.

Reply:
Good points, thanks!! I started a new discussion on the superceding PR here: https://github.com/lightningdevkit/rust-lightning/pull/1835#discussion_r1015967743. 

Re: (1) I'm not sure they could force us to store a ton of stuff since the payments would presumably be failed back as they come in by users that don't support intercept scid forwards (maybe the docs on the event should clarify that you MUST handle it)?